nexus-cancelation/caller/workflows.go

```diff
@@ -0,0 +1,75 @@
+package caller
+
+import (
+	"errors"
+
+	"github.com/temporalio/samples-go/nexus/service"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	TaskQueue    = "my-caller-workflow-task-queue"
+	endpointName = "my-nexus-endpoint-name"
+)
+
+var languages = []service.Language{
+	service.EN,
+	service.FR,
+	service.DE,
+	service.ES,
+	service.TR,
+}
+
+type result struct {
+	lang   service.Language
+	output service.HelloOutput
+	err    error
+}
+
+func HelloCallerWorkflow(ctx workflow.Context, name string) (string, error) {
+	log := workflow.GetLogger(ctx)
+	c := workflow.NewNexusClient(endpointName, service.HelloServiceName)
+	results := make([]result, len(languages))
+	callCtx, cancel := workflow.WithCancel(ctx)
+	defer cancel()
+
+	// Concurrently execute an operation per language.
+	wg := workflow.NewWaitGroup(ctx)
+	for i, lang := range languages {
+		wg.Add(1)
+		workflow.Go(ctx, func(ctx workflow.Context) {
+			defer wg.Done()
+			// Use the cancelable callCtx for executing the operation.
+			fut := c.ExecuteOperation(
+				callCtx,
+				service.HelloOperationName,
+				service.HelloInput{Name: name, Language: lang},
+				workflow.NexusOperationOptions{})
+			var output service.HelloOutput
+			err := fut.Get(ctx, &output)
+			results[i] = result{
+				lang:   lang,
+				output: output,
+				err:    err,
+			}
+			// The first operation to win the race cancels the others.
+			cancel()
```

Comment:
This is a bit different then the Java sample, there we waited for everything to start before cancelling

Reply:
That doesn't matter though, since the server will wait for start for us.
You can do the same in Java.

Reply:
I don't think the sample should be written assuming a generic async operation and not rely on specific behaviour of a workflow run

Reply:
The server doesn't know this is a workflow run operation, just that it's async (as indicated by the token returned from the handler).